framework/src/media/audio/audio_manager.c

```diff
@@ -453,22 +454,23 @@ static audio_manager_result_t get_audio_volume(int fd, audio_config_t *config, a
 		caps_desc.caps.ac_format.hw = AUDIO_FU_INP_GAIN;
 	}
 
-	ret = ioctl(fd, AUDIOIOC_GETCAPS, (unsigned long *)&caps_desc.caps);
-	if (ret < 0) {
-		meddbg("Fail to ioctl AUDIOIOC_GETCAPS\n");
-		return AUDIO_MANAGER_DEVICE_FAIL;
+	if ((ioctl(fd, AUDIOIOC_GETCAPS, (unsigned long *)&caps_desc.caps)) < 0) {
```

Comment:
This is not good, as per we discussed, below logic should be called if return value of ioctl is -ENOSYS

Reply:
Also, will you take policy that return NOT_SURRPOT for both of get & set?? There's chance that modify it later, I guess
Because we have to modify volume level in future based on decibel value

Reply:
Or you can provide API like a isVolumeFixed in android.
https://developer.android.com/reference/android/media/AudioManager.html#isVolumeFixed()

Reply:
You're right. In fact, I tried to set the volume only when -ENOSYS is returned, but the return value of ioctl is a bit weird.
I confirmed that -ENOSYS is returned in audio.c, but the return value of ioctl is not -ENOSYS.
Therefore I removed "if (ret == -ENOSYS)"..